torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,253 @@
+    assessment algorithms, which in turn are intended to compute the percentage of cloud cover in each scene. *Originally, this was reported as 207 masks, but one mask 
+    was erroneously counted.
+
+    Dataset format:
+    * 102 scenes from this dataset were used in a cloud validation study described in Foga, Scaramuzza et al. (2017)
+    * labels are values [0, 64, 128, 192, 255]
+
+     Dataset classes:
+     * Fill
+     * Cloud Shadow
+     * Clear
+     * Thin Cloud
+     * Cloud
+
+     If you use this dataset in your research, please cite the following papers:
+
+     * U.S. Geological Survey, 2016. L7 Irish Cloud Validation Masks. U.S. Geological Survey data release. doi:10.5066/F7XD0ZWC.
+     * Scaramuzza, P.L., Bouchard, M.A. & Dwyer, J.L. (2012). Development of the Landsat data continuity mission cloud-cover assessment algorithms. IEEE Transactions on Geoscience and Remote Sensing, 50(4), 1140-1154. doi:10.1109/TGRS.2011.2164087.
+     """
+
+    url = "https://huggingface.co/datasets/torchgeo/l7irish"
+    
+    md5 = ["dbb6b5628f50861b9b89f548d25a925f",
+           "cecc72de09aacde4c4f8d7f0cf0d3f6f",
+           "0f8382ca6554fb7cf9aff42226a14f9d",
+           "b17cf6d023f752c533211fdb742f296b",
+           "73923dcaf1b9b79bad82de1aa0740d1e",
+           "3bc9f4c6f8955b10b4d55d23e0ab2da7",
+           "f8f039970256902e6e9ebd6747589294",
+           "8346d73a983396c5d41b577c3a94bc26",
+           "abe19b22b5d031e6b609cc7207706c3d"]
+    
+    biomes = ["austral","boreal","mid_latitude_north","mid_latitude_south",
+              "polar_north","polar_south","subtropical_north",
+              "subtropical_south","tropical"]
+    
+    classes = ["Fill", "Cloud Shadow", "Clear", "Thin Cloud", "Cloud"]
+    
+    cmap = {
+        0: (0,0,0),
+        64: (64,64,64),
+        128: (128,128,128),
+        192: (192,192,192),
+        255: (255,255,255),
+    }
+
+    def __init__(
+        self, root: str = "data", download: bool = False, checksum: bool = False
+    ) -> None:
+        
+        """Initialize a new Landsat 7 Cloud Cover Assessment Validation dataset instance.
+
+            Args:
+                root: root directory where dataset can be found
+                transforms: a function/transform that takes input sample and its target as
+                    entry and returns a transformed version
+                cache: if True, cache file handle to speed up repeated sampling
+                download: if True, download dataset and store it in the root directory
+                checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+            Raises:
+                RuntimeError: if ``download=False`` and data is not found, or checksums
+                    don't match
+            """
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+        super().__init__(root, download, checksum)
```

Comment:
The base class doesn't have a download or checksum argument, this is wrong.

Reply:
Still wrong

Reply:
Indeed, I change the conversion method to zfill().